client/packages/system/src/Patient/Components/PatientSearchInput/PatientSearchInput.tsx

```diff
@@ -16,26 +16,53 @@ export const PatientSearchInput: FC<NameSearchInputProps> = ({
 }) => {
   const PatientOptionRenderer = getPatientOptionRenderer();
   const { isLoading, patients, search } = useSearchPatient();
+  const t = useTranslation();
+
+  const [input, setInput] = useState('');
+
+  useEffect(() => {
+    if (value) setInput(value.name);
+  }, [value]);
```

Comment:
Looking good -- there's just this one UI thing that I think we can quickly fix.

Seems like the patient options list is lingering with a stale value -- when you go into the input, the dropdown first shows the previous results (or no results) rather than properly reflecting the current selection:

![2025-01-29 16 33 51](https://github.com/user-attachments/assets/a5464444-0ead-4cca-9ad2-8d043da12be7)

This seems to fix it:
```ts
  useEffect(() => {
    if (value) {
      setInput(value.name);
      search(value.name);
    }
  }, [value]);
```

Bit clunky to run another search after a selection, but is there a better way?

Reply:
Done :) [0cd486b](https://github.com/msupply-foundation/open-msupply/pull/6328/commits/0cd486b511d9d4a7bd495dbf18527f46d85e2d8e)

Reply:
Realised the onFocus call wasn't needed with this change 🙏  [59e7782](https://github.com/msupply-foundation/open-msupply/pull/6328/commits/59e77821dd21212090b13a3f77b3ed57eea36d35)